plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java

```diff
@@ -102,13 +129,53 @@ public DockerConnectorConfiguration(URI dockerDaemonUri,
         this.networkFinder = networkFinder;
     }
 
-    public static String getExpectedLocalHost() {
-        return SystemInfo.isLinux() ? "localhost" : dockerDaemonUri().getHost();
+    public static boolean isUnixSocketUri(URI uri) {
```

Comment:
I think you should use `@NotNull` + `Objects.requireNonNull(uri, "Required non-null uri")` and write javadoc that explains NotNull contract, e.g. defines `@throws NullPointerException` with appropriate description.